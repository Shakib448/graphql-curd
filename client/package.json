{
  "name": "client",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "test": "jest"
  },
  "dependencies": {
    "@apollo/client": "^3.4.17",
    "@types/enzyme": "^3.10.10",
    "graphql": "^16.0.1",
    "next": "12.0.3",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "waait": "^1.0.5"
  },
  "jest": {
    "setupFilesAfterEnv": [
      "./src/setupTests.ts"
    ]
  },
  "devDependencies": {
    "@testing-library/dom": "^8.11.1",
    "@testing-library/jest-dom": "^5.15.0",
    "@testing-library/react": "^12.1.2",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.0.2",
    "@types/node": "^16.11.7",
    "@types/react": "^17.0.35",
    "@types/react-dom": "^17.0.11",
    "@types/react-test-renderer": "^17.0.1",
    "@typescript-eslint/eslint-plugin": "^5.3.1",
    "babel-jest": "^27.3.1",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.6",
    "eslint": "7",
    "eslint-config-next": "12.0.3",
    "jest": "^27.3.1",
    "jest-enzyme": "^7.1.2",
    "jsdom": "^18.1.0",
    "jsdom-global": "^3.0.2",
    "react-test-renderer": "^17.0.2",
    "ts-node": "^10.4.0",
    "typescript": "^4.4.4"
  }
}
