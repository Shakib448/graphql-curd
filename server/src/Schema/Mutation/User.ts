import { GraphQLString } from "graphql";
import { UserType } from "../TypeDefs/User";

export const CREATE_USER = {
  type: UserType,
  args: {
    name: { type: GraphQLString },
    username: { type: GraphQLString },
    password: { type: GraphQLString },
  },
  resolve(parent: any, args: any) {
    const { name, username, password } = args;
    return args;
  },
};
