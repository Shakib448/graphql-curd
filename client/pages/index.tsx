import React, { useState } from "react";
import Head from "next/head";
import { useMutation } from "@apollo/client";
import Link from "next/link";

import { CREATE_USER } from "../src/Graphql/Mutation";

const Home = () => {
  const [name, setName] = useState("");
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [createUser, { error, loading }] = useMutation(CREATE_USER);

  if (loading) return "Submitting...";
  if (error) return `Submission error! ${error.message}`;

  return (
    <div>
      <Head>
        <title>Create Next App</title>
        <meta name="description" content="Generated by create next app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main>
        <div className="createUser">
          <input
            type="text"
            placeholder="name"
            onChange={(e) => setName(e.target.value)}
          />
          <input
            type="text"
            placeholder="username"
            onChange={(e) => setUsername(e.target.value)}
          />
          <input
            type="text"
            placeholder="password"
            onChange={(e) => setPassword(e.target.value)}
          />
          <button
            onClick={() =>
              createUser({
                variables: {
                  name: name,
                  username: username,
                  password: password,
                },
              })
            }
          >
            submit
          </button>
          <Link href="/newUser">GO</Link>
        </div>
      </main>
    </div>
  );
};

export default Home;
