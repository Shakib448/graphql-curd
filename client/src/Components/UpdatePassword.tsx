import React, { useState } from "react";
import Head from "next/head";
import { useMutation } from "@apollo/client";
import { UPDATE_PASSWORD } from "../Graphql/Mutation";

const UpdatePassword = () => {
  const [username, setUsername] = useState("");
  const [oldPassword, setOldPassword] = useState("");
  const [newPassword, setNewPassword] = useState("");
  const [updatePassword, { error, loading }] = useMutation(UPDATE_PASSWORD);

  if (loading) return "Submitting...";
  if (error) return `Submission error! ${error.message}`;

  return (
    <div>
      <div className="createUser">
        <input
          type="text"
          placeholder="username"
          onChange={(e) => setUsername(e.target.value)}
        />
        <input
          type="text"
          placeholder="old password"
          onChange={(e) => setOldPassword(e.target.value)}
        />
        <input
          type="text"
          placeholder="new password"
          onChange={(e) => setNewPassword(e.target.value)}
        />
        <button
          onClick={() =>
            updatePassword({
              variables: {
                username: username,
                oldPassword: oldPassword,
                newPassword: newPassword,
              },
            })
          }
        >
          update password
        </button>
      </div>
    </div>
  );
};

export default UpdatePassword;
