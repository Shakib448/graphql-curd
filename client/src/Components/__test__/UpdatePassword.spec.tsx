/* eslint-disable testing-library/no-wait-for-snapshot */
/* eslint-disable testing-library/await-async-query */
import { render, waitFor } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { UPDATE_PASSWORD } from "../../Graphql/Mutation";
import userEvent from "@testing-library/user-event";
import UpdatePassword from "../UpdatePassword";

describe("Testing with renders", () => {
  it("renders with loading", () => {
    const mocks: any[] = [];
    const { container, getByText, getByRole } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <UpdatePassword />
      </MockedProvider>
    );

    const button = getByRole("button", { name: /update password/i });

    userEvent.click(button);

    const SubmittingMessage = getByText(/Submitting.../i);
    expect(SubmittingMessage).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("should delete with success message", async () => {
    const updateMessage = { message: "Successfully Delete" };
    const deleteMockUser: any = {
      request: {
        query: UPDATE_PASSWORD,
        variables: {
          username: "shakib",
          oldPassword: "muktadir",
          newPassword: "123456",
        },
      },
      result: {
        data: {
          updateMessage,
        },
      },
    };

    const { container, getByRole, getByText } = render(
      <MockedProvider mocks={[deleteMockUser]} addTypename={false}>
        <UpdatePassword />
      </MockedProvider>
    );

    const button = getByRole("button", { name: /update password/i });
    userEvent.click(button);

    await waitFor(async () => {
      const successDeleteMessage = getByText(/Successfully submitted!/i);
      expect(successDeleteMessage).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
