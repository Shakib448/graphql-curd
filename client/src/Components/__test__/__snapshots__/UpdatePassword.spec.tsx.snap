// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with renders renders with loading 1`] = `
<div>
  <h1>
    Submitting...
  </h1>
</div>
`;
